src/inmanta/deploy/scheduler.py

```diff
@@ -263,13 +264,26 @@ async def _new_version(
                 self._work.delete_resource(resource)
 
             new_desired_state: set[ResourceIdStr] = set()
+            # Only contains the direct undeployable resources, not the transitive ones.
+            blocked_resources: set[ResourceIdStr] = set()
+            # Resources that were undeployable in a previous model version, but got unblocked. Not the transitive ones.
+            unblocked_resources: set[ResourceIdStr] = set()
             added_requires: dict[ResourceIdStr, Set[ResourceIdStr]] = {}
             dropped_requires: dict[ResourceIdStr, Set[ResourceIdStr]] = {}
             for resource, details in resources.items():
-                if (
-                    resource not in self._state.resources
-                    or details.attribute_hash != self._state.resources[resource].attribute_hash
-                ):
+                if details.status is const.ResourceState.undefined:
```

Comment:
I see your point. It's rather confusing. The two types represent different things, but they have an almost identical name. Sticking with one will not be easy. But given that `ResourceStatus`  was introduced recently, I think we can still quite easily rename it. Maybe we should call it `ResourceOpStatus`, to indicate this represents the operational state of the resource. Do you have something else in mind? @sanderr I think you introduced the `ResourceStatus`. Do you have an opinion on this?

Reply:
I have to go through the full pull request before I can give a definite answer, but for now I'll say this:
1. I wouldn't expect `const.ResourceStatus` to be used here, because that is the legacy status that we're trying to replace here.
2. I'm open to a name change, and perhaps we can even drop the resource part and go for simply `OperationalStatus` but I'm not sure we can make that decision here. I would involve the people that were present for the [improved resource states design session](https://docs.google.com/presentation/d/1F3bFNy2BZtzZgAxQ3Vbvdw7BWI9dq0ty5c3EoLAtUUY/edit#slide=id.g292b508a90d_0_5). Iirc that's Bart, Wouter and Guillaume. I think this PR is the final piece in the resource state that we want to include for iso8. Perhaps, once review has settled, we can have a short session where we propose the subset of the improved resource states that we want to expose, with the names we intend to use, and only then make resulting changes?

Reply:
For now, I expect both to be used (one towards the DB and one towards the scheduler). Once the DB is more evolved, the `const.ResourceState.undefined` will go away 

We could raw the line a bit higher up on the stack, but this fine for me as well